src/mozanalysis/metrics/desktop.py

```diff
@@ -27,6 +29,32 @@
     experiments_column_type='native',
 )
 
+main = DataSource(
+    name='main',
+    from_expr=dedent("""\
+        (
+            SELECT
```

Comment:
In how many more places are we likely to see this pattern of `SELECT *` and turn a timestamp to a date and pull `experiments` out of the environment? Is it worthwhile supporting it in the `DataSource` class so that these `from_expr` could just become table names?

My guess right now is that this would be over-optimising, but dunno?

Reply:
Hm, thinking out loud: all of the pre-aggregation ping tables, including the Glean tables, but otoh there just aren't that many doctypes. The experiments column is in different places in different (e.g. Glean) tables, so we'd have to add a property to the DataSource class to specify that...

I don't really like asking consumers to code against a table spec that they can't see, so maybe it makes sense to introduce some concept of "native" and "aggregated" table types?

Reply:
> The experiments column is in different places in different (e.g. Glean) tables, so we'd have to add a property to the DataSource class to specify that...

We already have a property that does something very similar to that...

> I don't really like asking consumers to code against a table spec that they can't see, so maybe it makes sense to introduce some concept of "native" and "aggregated" table types?

I'm going to ask you in person what you mean by this.

Reply:
captured in #68